packages/text-field/Input.tsx

```diff
@@ -130,19 +130,21 @@ export default class Input<T extends HTMLElement = HTMLInputElement> extends Rea
       setInputId && setInputId(id!);
     }
 
-    if (value !== prevProps.value) {
+    if ((value !== prevProps.value || prevProps.foundation !== foundation) && foundation) {
       handleValueChange && handleValueChange(value, () => {
         // only call #foundation.setValue on programatic changes;
-        // not changes by the user.
-        if (this.state.wasUserTriggeredChange) {
+        // not changes by the user. or when we go from no foundation
+        // to having a foundation, since we don't have that guaranteed
+        // anymore
+        if (this.state.wasUserTriggeredChange && prevProps.foundation === foundation) {
```

Comment:
I find this confusing, since right above this we are checking to see if the foundation has changed. Is there some way to make this a little more clear?

Reply:
I think it's a bit more clear now, with the variable name for the condition